src/browser/BrowserCbor.cpp

```diff
@@ -0,0 +1,253 @@
+    writer.appendByteString(authData.constData(), authData.size());
+
+    writer.endMap();
+
+    return result;
+}
+
+// https://w3c.github.io/webauthn/#authdata-attestedcredentialdata-credentialpublickey
+QByteArray BrowserCbor::cborEncodePublicKey(int alg, const QByteArray& first, const QByteArray& second) const
+{
+    QByteArray result;
+    QCborStreamWriter writer(&result);
+
+    if (alg == WebAuthnAlgorithms::ES256) {
+        writer.startMap(5);
+
+        // Key type
+        writer.append(1);
+        writer.append(getCoseKeyType(alg));
+
+        // Signature algorithm
+        writer.append(3);
+        writer.append(alg);
+
+        // Curve parameter
+        writer.append(-1);
+        writer.append(getCurveParameter(alg));
+
+        // Key x-coordinate
+        writer.append(-2);
+        writer.append(first);
+
+        // Key y-coordinate
+        writer.append(-3);
+        writer.append(second);
+
+        writer.endMap();
+    } else if (alg == WebAuthnAlgorithms::RS256) {
+        writer.startMap(4);
+
+        // Key type
+        writer.append(1);
+        writer.append(getCoseKeyType(alg));
+
+        // Signature algorithm
+        writer.append(3);
+        writer.append(alg);
+
+        // Key modulus
+        writer.append(-1);
+        writer.append(first);
+
+        // Key exponent
+        writer.append(-2);
+        writer.append(second);
+
+        writer.endMap();
+    } else if (alg == WebAuthnAlgorithms::EDDSA) {
+        // https://www.rfc-editor.org/rfc/rfc8152#section-13.2
+        writer.startMap(3);
+
+        // Curve parameter
+        writer.append(-1);
+        writer.append(getCurveParameter(alg));
+
+        // Public key
+        writer.append(-2);
+        writer.append(first);
+
+        // Private key
+        writer.append(-4);
+        writer.append(second);
+
+        writer.endMap();
+    }
+
+    return result;
+}
+
+// See: https://fidoalliance.org/specs/common-specs/fido-registry-v2.1-ps-20191217.html#user-verification-methods
+QByteArray BrowserCbor::cborEncodeExtensionData(const QJsonObject& extensions) const
+{
+    if (extensions.empty()) {
+        return {};
+    }
+
+    QByteArray result;
+    QCborStreamWriter writer(&result);
+
+    writer.startMap(extensions.keys().count());
+    if (extensions["credProps"].toBool()) {
+        writer.append("credProps");
+        writer.startMap(1);
+        writer.append("rk");
+        writer.append(true);
+        writer.endMap();
+    }
+
+    if (extensions["uvm"].toBool()) {
+        writer.append("uvm");
+
+        writer.startArray(1);
+        writer.startArray(3);
+
+        // userVerificationMethod (USER_VERIFY_PRESENCE_INTERNAL "presence_internal", 0x00000001)
+        writer.append(1);
+
+        // keyProtectionType (KEY_PROTECTION_SOFTWARE "software", 0x0001)
+        writer.append(1);
+
+        // matcherProtectionType (MATCHER_PROTECTION_SOFTWARE "software", 0x0001)
+        writer.append(1);
```

Comment:
Does it matter that presence_internal is 8-bit (based on the comment) and the others are 4-bit?

Reply:
The debugger site doesn't complain about this, but let's change these just to make sure.

Reply:
Geez even I screwed up the bit count. Hexadecimal before coffee...